From 8867d949cdc9a786965d95856cb145341d3185ab Mon Sep 17 00:00:00 2001
From: fupan li <fupan.li@windriver.com>
Date: Wed, 18 Nov 2015 15:57:40 +0800
Subject: [PATCH] systemd: make some services container awared

For modules-load and random-seed services, they are unnecessary
for containers init, so make them container awared and not run
in it.

By now, overc containers only copied the host's dev nodes and
didn't mount the devtmpfs, udevd was useless in it and make them
container awared and not run in it. Maybe in the future udevd is used
when supporting container's device hotplug and we can remove this
for udev.

Signed-off-by: fupan li <fupan.li@windriver.com>
---
 units/systemd-modules-load.service.in | 1 +
 units/systemd-random-seed.service.in  | 1 +
 units/systemd-udev-trigger.service.in | 1 +
 units/systemd-udevd-control.socket    | 1 +
 units/systemd-udevd-kernel.socket     | 1 +
 units/systemd-udevd.service.in        | 1 +
 6 files changed, 6 insertions(+)

diff --git a/units/systemd-modules-load.service.in b/units/systemd-modules-load.service.in
index 32deb52..99a19ee 100644
--- a/units/systemd-modules-load.service.in
+++ b/units/systemd-modules-load.service.in
@@ -9,6 +9,7 @@
 Description=Load Kernel Modules
 Documentation=man:systemd-modules-load.service(8) man:modules-load.d(5)
 DefaultDependencies=no
+ConditionVirtualization=no
 Conflicts=shutdown.target
 After=systemd-readahead-collect.service systemd-readahead-replay.service
 Before=sysinit.target shutdown.target
diff --git a/units/systemd-random-seed.service.in b/units/systemd-random-seed.service.in
index 1879b2f..041d784 100644
--- a/units/systemd-random-seed.service.in
+++ b/units/systemd-random-seed.service.in
@@ -9,6 +9,7 @@
 Description=Load/Save Random Seed
 Documentation=man:systemd-random-seed.service(8) man:random(4)
 DefaultDependencies=no
+ConditionVirtualization=no
 RequiresMountsFor=@RANDOM_SEED@
 Conflicts=shutdown.target
 After=systemd-readahead-collect.service systemd-readahead-replay.service systemd-remount-fs.service
diff --git a/units/systemd-udev-trigger.service.in b/units/systemd-udev-trigger.service.in
index 0c33909..25dc8f7 100644
--- a/units/systemd-udev-trigger.service.in
+++ b/units/systemd-udev-trigger.service.in
@@ -9,6 +9,7 @@
 Description=udev Coldplug all Devices
 Documentation=man:udev(7) man:systemd-udevd.service(8)
 DefaultDependencies=no
+ConditionVirtualization=no
 Wants=systemd-udevd.service
 After=systemd-udevd-kernel.socket systemd-udevd-control.socket
 Before=sysinit.target
diff --git a/units/systemd-udevd-control.socket b/units/systemd-udevd-control.socket
index 8330a1c..3bdc6e8 100644
--- a/units/systemd-udevd-control.socket
+++ b/units/systemd-udevd-control.socket
@@ -9,6 +9,7 @@
 Description=udev Control Socket
 Documentation=man:systemd-udevd.service(8) man:udev(7)
 DefaultDependencies=no
+ConditionVirtualization=no
 Before=sockets.target
 ConditionPathIsReadWrite=/sys
 
diff --git a/units/systemd-udevd-kernel.socket b/units/systemd-udevd-kernel.socket
index 39b7809..fa626c7 100644
--- a/units/systemd-udevd-kernel.socket
+++ b/units/systemd-udevd-kernel.socket
@@ -9,6 +9,7 @@
 Description=udev Kernel Socket
 Documentation=man:systemd-udevd.service(8) man:udev(7)
 DefaultDependencies=no
+ConditionVirtualization=no
 Before=sockets.target
 ConditionPathIsReadWrite=/sys
 
diff --git a/units/systemd-udevd.service.in b/units/systemd-udevd.service.in
index f6acd6f..7275e69 100644
--- a/units/systemd-udevd.service.in
+++ b/units/systemd-udevd.service.in
@@ -9,6 +9,7 @@
 Description=udev Kernel Device Manager
 Documentation=man:systemd-udevd.service(8) man:udev(7)
 DefaultDependencies=no
+ConditionVirtualization=no
 Wants=systemd-udevd-control.socket systemd-udevd-kernel.socket
 After=systemd-udevd-control.socket systemd-udevd-kernel.socket systemd-udev-hwdb-update.service systemd-sysusers.service
 Before=sysinit.target
-- 
2.4.5

